torch/utils/data/dataset.py

```diff
@@ -251,6 +251,31 @@ def __getitem__(self, index):
             return {k: dataset[index] for k, dataset in self.datasets.items()}
         return tuple(dataset[index] for dataset in self.datasets)
 
+    def __getitems__(self, indices: list):
+        # add batched sampling support when parent datasets supports it.
+        if isinstance(self.datasets, dict):
+            dict_batch: List[T_dict] = [{} for _ in indices]
+            for k, dataset in self.datasets.items():
+                if callable(getattr(dataset, "__getitems__", None)):
+                    for data, d_sample in zip(dataset.__getitems__(indices), dict_batch):  # type: ignore[attr-defined]
```

Comment:
Could you pls add an Error if the size of result from `dataset.__getitems__(indices)` is not correct?